introduce more automation for dedicated Hetzner server deployments prior to kubespray upgrades (#676)
wireapp/wire-server-deploy: offline/single_hetzner_machine_installation.md

This document gives exact instructions for performing an offline installation of Wire on a single VM from Hetzner. it uses the KVM based virtual machine system to create all of the required virtual machines.

Bootstrapping a single dedicated Hetzner server for virtual machine deployment as well as wire-server-deploy artifact download has largely been automated with ansible and bash.

## Use the hetzner robot console to create a new server.

> Select Ubuntu 22.04.2 on an ax101 dedicated server. If possible, please already provide a public key in the Hetzner console which can be used for ansible deployment.

Reviewer: ```suggestion
Select Ubuntu 22.04.2 on an ax101 dedicated server. Make sure you provide a public key in the Hetzner console which can be used for ansible deployment.
```

- 2 x 3.84 TB NVMe SSD Datacenter Edition (software RAID 1)
- 1 GBit/s port

Please note the public IP of the newly provisioned server, as it's used for the ansible playbook run.

## Pre-requisites (ansible)

Make sure to have ansible available on your local computer with a recent version (eg. 2.14.9 as of 2024-01).

## Adjust playbook vars as needed

Take a look at the "vars:" section in wire-server-deploy/ansible/hetzner-single-deploy.yml and adjust vars as needed. Example:
```
  vars:
    artifact_hash: a6e0929c9a5f4af09655c9433bb56a4858ec7574
    ubuntu_version: 22.04.3

>     ssh_pubkey: "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIODDzgw4BncNvSVOIraAXZYkkLn+iTo6ixkXXQ4lKZhL jan.schumacher@wire.com"

Reviewer 2: might want to just change the key comment here and consider not using a valid key. i am sure a user might accidentally copy this one day :)

- 2 x 3.84 TB NVMe SSD Datacenter Edition (software RAID 1)
- 1 GBit/s port

> Please note the public IP of the newly provisioned server, as it's used for the ansible playbook run.

Reviewer: we're both wrong. tell the user where to get the IP in the interface.

## Adjust ansible playbook vars as needed

> Take a look at the "vars:" section in wire-server-deploy/ansible/hetzner-single-deploy.yml and adjust vars as needed. Example:

Reviewer: traditionally, we don't edit anything but the hosts.ini, in ansible/.

"
```

## Access VMs

> VMs created with offline-vm-setup.sh are accessible via SSH with two public keys.

Reviewer: you might want to say where they're accessable from.
